### DISCUSSION TITLE
大佬看看我这是哪搞错了

### DISCUSSION THREAD

**Discussion by QAD21** [xiaohucode/yidaRule#200] [Q&A]
在网站控制台输出是有结果的，但是为啥在软件上提示错误没结果啊：
function findImageUrlsInThirdScript() {  
    var scriptTags = document.querySelectorAll('script'); 
    var thirdScript; 
  
    if (scriptTags.length > 2) { 
        thirdScript = scriptTags[2];  
    }  
  
    if (thirdScript) {  
        var matches = thirdScript.textContent.match(/https:\/\/[^\s'"]+\.jpg/g);
        if (matches) {  
            return matches;  
        } else {  
            return []; 
        }  
    }  
    return null; 
}  
var imageUrls = findImageUrlsInThirdScript();  

console.log(imageUrls);

![Screenshot_20240721_215718_com xinqiu yidaapp](https://github.com/user-attachments/assets/616d6cd0-74b1-47cf-b3fb-3c427553903e)

**Comment by chenxingmoonset** [comment]
从结果上看应该是不支持web api的，不如用内置的tools.cssSelector()方法试试？

**Comment by QAD21** [comment]
我改成了用xpath写了：    var scriptTags = tools.xpathSelector(result, '//script'); ，但是提示我：[contentRule.content]SyntaxError: expecting ';'
    at <eval>:2     这是什么原因啊，符号好像没少啊

**Reply by chenxingmoonset** [reply]
或许是没写await？
